Components/ListOfCharts/index.js

```diff
@@ -3,20 +3,25 @@ import { Container } from "../Container";
 import ChartItem from "../ChartItem";
 import { chartArray } from "../../lib/listOfPlotTypes";
 
-export default function ListOfCharts({ onSelectChartType, keynames }) {
+export default function ListOfCharts({
+  onSelectChartType,
+  keynames,
+  clickedChartType,
+}) {
   return (
     <>
       {keynames.length > 0 && (
         <>
           <Heading>Step 2: Select the type of chart</Heading>
           <Container $wrap="wrap">
-            {chartArray.map((chart) => {
+            {chartArray.map(({ name, icon }) => {
               return (
                 <ChartItem
-                  key={chart.name}
-                  icon={chart.icon}
-                  name={chart.name}
-                  onSelectChartType={onSelectChartType}
+                  key={name}
+                  icon={icon}
+                  name={name}
+                  clickedChartType={clickedChartType}
+                  onSelectChartType={() => onSelectChartType({ name })}
```

Comment:
why do you pass the name as an object?

If you were to pass it like this:
```suggestion
                  onSelectChartType={() => onSelectChartType(name)}
```

You would not need to destructure it in your `handleSelectChartType` function in your `_app.js`

Reply:
Done. Thank's a lot for the suggestions.